pkg/client/restclient/plugin.go

```diff
@@ -30,9 +30,16 @@ type AuthProvider interface {
 	// WrapTransport allows the plugin to create a modified RoundTripper that
 	// attaches authorization headers (or other info) to requests.
 	WrapTransport(http.RoundTripper) http.RoundTripper
+	// Login is expected to maybe require user interaction; it won't be called
+	// in environments where a user is not present
+	Login() error
```

Comment:
@deads2k: For the refresh case I don't think we need to add anything to the interface: The `RoundTripper` that is returned by `WrapTransport` should be able to handle that as necessary (eg, when short term creds are about to expire/have expired), as long as it can get a Persister. 

So I think we only need a method for getting the initial credentials, which I would think it's safe to assume, will always require some sort of user interaction.